docs/projects/automate-project-git-source.ipynb

```diff
@@ -0,0 +1,2069 @@
+{
```

Comment:
<p>with_source_archive can be used for nuclio,serving and jobs, please add a link to the same method for nuclio and serving functions - https://docs.mlrun.org/en/stable/api/mlrun.runtimes.html?highlight=with_source_archive#mlrun.runtimes.RemoteRuntime.with_source_archive</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3640/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Reply:
added link

Reply:
<p>If&nbsp;<code>with_repo</code>&nbsp;is not set to&nbsp;<code>True</code>&nbsp;(default), you need to deploy the functions with&nbsp;<code>function.deploy()</code>&nbsp;to build a</p><p>container - it is relevant to to pull_at_runtime flag need to change to </p><p>If&nbsp;<code>pull_at_runtime</code>&nbsp;is not set to&nbsp;<code>True</code>&nbsp;(default), you need to deploy the functions with&nbsp;<code>function.deploy()</code>&nbsp;to build a</p><p>container.</p><p>or you can use this paragraph <code> When using with_repo=True the functions need to be deployed (function.deploy()) to build a container, unless you set project.set_source(source=source,pull_at_runtime=True) which instructs MLRun to load the git/archive repo into the function container at run time and do not require a build (this is simpler when developing, for production it’s preferred to build the image with the code)</code></p><p> </p><p> </p>

Reply:
<p>updated, pls see</p>